### PR TITLE
Upgrade to 4.20

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Changes required to run on 420.
420 engine branch: https://github.com/improbableio/UnrealEngine/tree/4.20-SpatialOSUnrealGDK
Test suite PR: https://github.com/spatialos/UnrealGDKTestSuite/pull/112
TPS PR: https://github.com/spatialos/UnrealGDKThirdPersonShooter/pull/26

#### Tests
Ran TestSuite, and all tests ran.

#### Documentation
Documentation updated to point to the 420 branch.

#### Primary reviewers
@girayimprobable @danielimprobable

### CODE REVIEW

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -358,7 +358,7 @@ bool USpatialActorChannel::ReplicateActor()
 
 	bForceCompareProperties = false;		// Only do this once per frame when set
 
-	return bWroteSomethingImportant;
+	return (bWroteSomethingImportant) ? 1 : 0;	// TODO: return number of bits written
```
Comment (danielimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Jira ticket?

File: SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ void USpatialPackageMapClient::RemoveEntityActor(Worker_EntityId EntityId)
 	}
 }
 
-FNetworkGUID USpatialPackageMapClient::ResolveStablyNamedObject(const UObject* Object)
+FNetworkGUID USpatialPackageMapClient::ResolveStablyNamedObject(UObject* Object)
```
Comment (danielimprobable) [inline SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: How was this enforced by 4.20?

Comment (m-samiec) [reply SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: GetNetworkChecksum is now being used internally, and it doesn't take a const UObject.

File: SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ namespace
 // This is a bookkeeping function that is similar to the one in RepLayout.cpp, modified for our needs (e.g. no NaKs)
 // We can't use the one in RepLayout.cpp because it's private and it cannot account for our approach.
 // In this function, we poll for any changes in Unreal properties compared to the last time we replicated this actor.
-void UpdateChangelistHistory(FRepState * RepState)
+void UpdateChangelistHistory(TSharedPtr<FRepState> RepState)
```
Comment (Vatyx) [inline SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: `TSharedPtr<FRepState>&`?

Comment (danielimprobable) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: Doesn't & break the refcounting?

Comment (Vatyx) [reply SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp]: This function is just manipulating `RepState`, not storing a reference to it, so there's no need to increment the ref count.

File: SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -981,7 +981,7 @@ USpatialNetConnection* USpatialNetDriver::AcceptNewPlayer(const FURL& InUrl, boo
 		{
 			// Most of this is taken from "World->SpawnPlayActor", excluding the logic to spawn a pawn which happens during
 			// GameMode->PostLogin(...).
-			APlayerController* NewPlayerController = GameMode->SpawnPlayerController(ROLE_AutonomousProxy, FVector::ZeroVector, FRotator::ZeroRotator);
+			APlayerController* NewPlayerController = GameMode->SpawnPlayerController(ROLE_AutonomousProxy, TEXT(""));
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: This is awful.

Comment (m-samiec) [reply SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: Agreed.

Comment (Vatyx) [reply SpatialGDK/Source/Private/EngineClasses/SpatialNetDriver.cpp]: Could we put anything meaning full for that string? I have no idea what the parameters are.

File: SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp [hunk 5]
```diff
@@ -111,7 +111,7 @@ FString PropertyToSchemaType(UProperty* Property, bool bIsRPCProperty)
 
 void WriteSchemaRepField(FCodeWriter& Writer, const TSharedPtr<FUnrealProperty> RepProp, const FString& PropertyPath, const int FieldCounter)
 {
-	Writer.Printf("%s %s = %d; // %s // %s",
+	Writer.Printf("{0} {1} = {2}; // {3} // {4}",
```
Comment (Vatyx) [inline SpatialGDKEditorToolbar/Source/Private/SchemaGenerator/SchemaGenerator.cpp]: :OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO NICE

**Comment by m-samiec** [comment]
> Does Starter Project need a PR too?

No changes needed for StarterProject.

**Pull request merged**